activestorage/app/javascript/activestorage/ujs.js

```diff
@@ -15,9 +15,9 @@ export function start() {
 }
 
 function didClick(event) {
-  const { target } = event
-  if ((target.tagName == "INPUT" || target.tagName == "BUTTON") && target.type == "submit" && target.form) {
-    submitButtonsByForm.set(target.form, target)
+  const submitButton = event.target.closest(":is(button, input)[type='submit']")
+  if (submitButton && submitButton.form) {
```

Comment:
This selector won’t match `<button>`s with an implicit type of `"submit"`:
```html
<form>
  <button id="x">…</button>
</form>
```
```js
> document.querySelector(":is(button, input)[type='submit']")
null

> document.getElementById("x").type
"submit"
```

Suggestion:
```js
const submitButton = event.target.closest("input, button")
if (submitButton && submitButton.type == "submit" && submitButton.form) {
```

Reply:
Fixed in #50165.  Thank you, @javan! :bow: